rfcs/0038-cheque/0038-cheque.md

```diff
@@ -0,0 +1,332 @@
+
+     - 2.b.i. It loops through all input cells using the receiver lock hash, if the matching inputs are not found, the cheque lock returns with an error state
+
+     - 2.b.ii It loops through all input cells using the receiver lock hash, if the first witness of the cheque cell group is empty(the witness is not [WitnessArgs](https://github.com/nervosnetwork/ckb/blob/a6733e6af5bb0da7e34fb99ddf98b03054fa9d4a/util/types/schemas/blockchain.mol#L104-L108) or the lock of WitnessArgs is empty)
+
+   - 2.c. If the matching input cells with the sender lock hash are found, it performs the same check as in 1.c.i.
+
+     Notice the cheque lock script includes a public key hash, two cheque lock scripts using the same cheque lock code but different public key hash, will be treated as different lock scripts, and each will perform the script unlock rule checking independently.
+
+## Examples
+
+### Create a cheque Cell
+
+```
+Inputs:
+    SUDT Cell:
+        Capacity: 1000 CKBytes
+        Lock: <sender_secp256k1_blake2b_lock_script>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 1000 UDT
+
+Outputs:
+    Cheque Cell:
+        Capacity: 165 CKBytes
+        Lock:
+            code_hash: cheque lock
+            args:  <receiver_secp256k1_blake2b_lock_hash[0..20]> <sender_secp256k1_blake2b_lock_hash[0..20]>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 100 UDT
+    SUDT Cell:
+        Capacity: 834.99 CKBytes
+        Lock: <sender_secp256k1_blake2b_lock_script>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 900 UDT
+
+Witnesses
+    <valid signature for sender public key hash>
+```
+
+In this example 0.01 CKBytes is paid as the transaction fee, in production one should calculate the fee based on factors including transaction size, running cycles as well as network status. 0.01 CKByte will be used in all examples as fees for simplicity.
+
+### Claim
+
+#### 1. Claim via receiver signature
+
+```
+Inputs:
+    Cheque Cell:
+        Capacity: 165 CKBytes
+        Lock:
+            code_hash: cheque lock
+            args: <receiver_secp256k1_blake2b_lock_hash[0..20]> <sender_secp256k1_blake2b_lock_hash[0..20]>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 100 UDT
+    SUDT Cell:
+        Capacity: 200 CKBytes
+        Lock: <another_receiver_lock_script>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 200 UDT
+
+ Outputs :
+    SUDT Cell :
+        Capacity: 199.99 CKBytes
+        Lock: <another_receiver_lock_script>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 300 UDT
+    Empty Cell:
+        Capacity: 165 CKBytes
+        Lock: <sender_secp256k1_blake2b_lock_script>
+
+ Witnesses :
+      < valid signature for another public key hash >
+      < valid signature for receiver public key hash >
```

Comment:
I think the witnesses should be(the order is not rignt?):
```
Witnesses :
      < valid signature for receiver public key hash >
      < valid signature for other public key hash >
```

Reply:
Updated.